jellyfin.ruleset

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="Rules for Emby.AutoOrganize" Description="Code analysis rules for Emby.AutoOrganize.csproj" ToolsVersion="14.0">
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
+    <!-- disable warning CA1032: Implement standard exception constructors -->
+    <Rule Id="CA1032" Action="Info" />
+    <!-- disable warning SA1202: 'public' members must come before 'private' members -->
+    <Rule Id="SA1202" Action="Info" />
+    <!-- disable warning SA1204: Static members must appear before non-static members -->
+    <Rule Id="SA1204" Action="Info" />
+    <!-- disable warning CA2227: Collection properties should be read only -->
+    <Rule Id="CA2227" Action="Info" />
```

Comment:
This file is a copy of the one from the main `jellyfin` repository, with the single addition of disabling `CA2227`.
Collection properties are mostly used in DTO objects in this project, but they are not deserialized correctly when set to read-only, so I just disabled this rule.

Reply:
What changes did you try, and what broke exactly?

Reply:
The initial steps I took to make the collections read-only were:
- Remove the setters on all collection properties
- Update code assigning values to the setters with `Clear()`/`AddRange()` where necessary
- Set default values for the collections to an empty collection (list/dictionary)

However, that seemed to break any of the web requests using those collections. Couple examples I noticed:
- In `Emby.AutoOrganize\Api\FileOrganizationService.cs`, `DeleteSmartMatchEntry.Entries` was always an empty list when received in the associated web request
- In `Emby.AutoOrganize\Model\TvFileOrganizationOptions.cs`, `LeftOverFileExtensionsToDelete` and `WatchLocations` were both always received as an empty list when updating settings

I know this type of deserialization is possible with the built in JSON serializer or with Json.NET, but I couldn't find any documentation on how to do it with ServiceStack. Any tips on how to make it work?

Reply:
It’s fine, thanks for the detailed explanation.